### DISCUSSION TITLE
macOS dependency problem with berkeley-db

### DISCUSSION THREAD

**Discussion by glennbacon** [dogecoin/dogecoin#2382] [Help needed]
Try to build dogecoin on macOS Big Sur. I follow the [build instructions](https://github.com/dogecoin/dogecoin/blob/master/doc/build-osx.md) but the only brew install version for berkeley-db is 18.1.40 and MacPorts does not have version 5.1.29 either. I'm stuck at this point. I did try installing berkeley-db 5.1.29 from [https://noknow.info/it/os/install_oracle_berkeley_db_from_source?lang=en#sec1](https://noknow.info/it/os/install_oracle_berkeley_db_from_source?lang=en#sec1) but I think maybe dogecoin configure files do not like the directories. Any help would be great.

**Comment by AbcSxyZ** [comment]
Did you solved your problem ?
2 idea :
- You can use another version of berkeley-db using the `--with-incompatible-bdb`.
- If you compile bdb yourself, does your prefix are correct ? Specifying flags on configuration : `./configure LDFLAGS="-L${BDB_PREFIX}/lib/" CPPFLAGS="-I${BDB_PREFIX}/include/" # (other args...)`. If it's not solved, share your commands

**Reply by glennbacon** [reply]
I didn't solve the problem. I wanted to use a compatible version of bdb and I haven't tried the second option. At this point I'm just going to see if I can get it working on my Windows laptop.

**Reply by AbcSxyZ** [reply]
Perhaps if you just want to run a node you can download the windows version of a release : https://github.com/dogecoin/dogecoin/releases

You will have compiled binaries with bdb 5.1

**Comment by michilumin** [comment]
--with-incompatible-bdb isn't a solution. Glenn, I have an updated doc for Big Sur that I'll be merging soon that addresses this. Basically you pull bdb source and build it, vs getting it from macports. Doesn't take much longer but I have the locations in that doc which should be updated soon.

**Reply by glennbacon** [reply]
That sounds great. I look forward to it thank you.

**Comment by micaelmalta** [comment]
@glennbacon I worked on a PR to make it work: https://github.com/dogecoin/dogecoin/pull/2398

Please follow this doc: https://github.com/dogecoin/dogecoin/blob/4767d2d06e0783a0fd3ff9cb8c8b76366a18909f/doc/Building-Dogecoin-1.14-for-Mac.md#compile-bdb-51
